reVX/offshore/offshore_inputs.py

```diff
@@ -0,0 +1,300 @@
+
+    @property
+    def column_ids(self):
+        """
+        Input layer array column ids that correspond to desired offshore sites
+
+        Returns
+        -------
+        ndarray
+        """
+        return self.meta['col_id'].values
+
+    @staticmethod
+    def _reduce_tech_map(inputs_fpath, tm_dset='techmap_wtk',
+                         offshore_gids=None):
+        """
+        Find the row and column indices that correspond to the centriod of
+        each offshore gid in exclusions layers. If offshore gids are not
+        provided the centroid of every gid is in techmap.
+
+        Parameters
+        ----------
+        inputs_fpath : str
+            Path to offshore inputs .h5 file
+        tm_dset : str, optional
+            Dataset / layer name for wind toolkit techmap,
+            by default 'techmap_wtk'
+        offshore_gids : ndarray | list, optional
+            Vector or list of offshore gids, by default None
+
+        Returns
+        -------
+        tech_map : pandas.DataFrame
+            DataFrame mapping resource gid to exclusions latitude, longitude,
+            row index, column index
+        """
+        with ExclusionLayers(inputs_fpath) as f:
+            tech_map = f[tm_dset]
+
+        gids = np.unique(tech_map)
+
+        if offshore_gids is None:
+            offshore_gids = gids[gids >= 0]
+        else:
+            missing = ~np.isin(offshore_gids, gids)
+            if np.any(missing):
+                msg = ('The following offshore gids were requested but are '
+                       'not availabe in {} and will not be extracted:\n{}'
+                       .format(tm_dset, offshore_gids[missing]))
+                logger.warning(msg)
+                warn(msg)
+                offshore_gids = offshore_gids[~missing]
+
+        tech_map = np.array(center_of_mass(tech_map, labels=tech_map,
+                                           index=offshore_gids),
+                            dtype=np.uint32)
+
+        tech_map = pd.DataFrame(tech_map, columns=['row_id', 'col_id'])
+        tech_map['gid'] = offshore_gids
+
+        return tech_map
+
+    @staticmethod
+    def _parse_offshore_sites(offshore_sites):
+        """
+        Load offshore sites from disc if needed
+
+        Parameters
+        ----------
+        offshore_sites : str | list | tuple | ndarray |pandas.DataFrame
+            - Path to .csv|.json file with offshore sites meta data
+            - Path to a WIND Toolkit .h5 file to extact site meta from
+            - List, tuple, or vector of offshore gids
+            - Pre-extracted site meta DataFrame
+
+        Returns
+        -------
+        offshore_sites : pandas.DataFrame
+            Offshore sites meta data
+        """
+        if isinstance(offshore_sites, str):
+            if offshore_sites.endswith('.h5'):
+                with Resource(offshore_sites) as f:
+                    offshore_sites = f.meta.reset_index()
```

Comment:
Is this how you're getting "gid" as a column? Kind of implicit, hard to read, subject to index naming conventions of the Resource handler.

Reply:
I'll make it more explicit